Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif

```diff
@@ -0,0 +1,139 @@
+    src_user2 = arrayindex(regextract(data -> username, "(.+)\@") ,0),
+    mod = if(data -> modified = "[]", null, data -> modified)
+| alter
+    xdm.event.operation_sub_type = coalesce(data -> dict_op, data -> action),
+    xdm.target.application.name = data -> applicationname,
+    xdm.event.description = if(data -> cause != null AND data -> modified != null, concat(rtrim(data -> cause, "."), ". modified properties: ", mod), coalesce(data -> cause, data -> modified)), // Checks if 'cause' and 'modified' are both populated. If so, appends the cause with a list of modified properties. If not, returns whichever field is available.
+    xdm.event.original_event_type = data -> messageid,
+    xdm.auth.mfa.method = data -> mfamethod,
+    xdm.source.user.identifier = if(data -> performedby IN("System", "shared_apikey"), null, data -> performedby),
+    xdm.source.application.name = coalesce(data -> performedby_clientname, if(data -> performedby IN("System", "shared_apikey"), data -> performedby)),
+    xdm.source.user.username = coalesce(src_user2, data -> username, src_user1, data -> performedby_username), //2 possible username fields, performedby_username and username.
+    xdm.source.user.upn = if(src_user2 != null, data -> username ,src_user1 != null, data -> performedby_username),
+    xdm.target.resource.id = data -> entitlementid,
+    xdm.target.resource.name = data -> entitlementname,
+    xdm.target.resource.sub_type = data -> entitlementsubtype,
+    xdm.target.resource.type = if(data -> entitlementtype != null AND data -> entitlementcategory != null, concat(data -> entitlementtype, " - ", data -> entitlementcategory), coalesce(data -> entitlementtype, data -> entitlementcategory, data -> resource )), // The "type" field represents the role, while the "category" field specifies the role type, e.g., "role - admin."
+    xdm.source.user.groups = to_json_string(data -> entitlements) -> []
+| call generic_fields_mapping;
+
+/*
+Mapping of 'sso' event types.
+*/
+filter event_type = "sso"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.auth.service = data -> subtype,
+    xdm.source.user.identifier = data -> userid,
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.target.application.name = data -> applicationname,
+    xdm.source.host.device_id = data -> deviceid //valid only if its an Android or iPhone device.
+| call generic_fields_mapping;
+
+/*
+Mapping of 'authentication' event types
+*/
+filter event_type = "authentication"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.event.operation_sub_type = coalesce(data -> dict_op, data -> action),
```

Comment:
For xdm.event.operation_sub_type we aim to normalize into one of the following groups:
- voice
- sms
- email
- hardware_token 
- trusted_login
- applications
see examples here: https://gitlab.xdr.pan.local/xdr/analytics-research/analytics_content/-/merge_requests/3284/diffs#5654a978f9cf5fd2b142313a3f8bd61ace085a96

so in our case we can do something like (this is not the full mapping, need to dive deeper):
case
           when data.subtype in ("user_password") then "password"
           when data.mfamethod in ("FIDO2") then "hardware_token"
           when data.mfamethod in ("Email OTP") then "email"
           when data.subtype in ("saml") then "Generic SSO"
end  as xdm.event.operation_sub_type

Reply:
Need to be normalized like in Okta (https://gitlab.xdr.pan.local/xdr/analytics-research/analytics_content/-/merge_requests/3284/diffs#5f8e4b344d259c9bf1c1760dc8d2087687aa5113)
        # Email OTP: A one-time passcode sent via email.
        # SMS OTP: A one-time passcode sent via SMS to the user’s phone.
        # Voice callback OTP: A code delivered via a voice call to the user’s phone.
        # Time-based OTP (TOTP): A code generated by an authenticator app (such as Google Authenticator or IBM Verify).
        # IBM Verify App (user presence): A push notification sent to the IBM Verify mobile app that requires user confirmation.
        # IBM Verify App (biometric): A biometric challenge, such as fingerprint or facial recognition, via the IBM Verify app.
        # FIDO2: A hardware security key or biometric method compliant with the FIDO2 standard.